Doc-string coverage improved

GeVic opened the pull request:
# Description
Doc-string support for implemented functions and classes. 

Fixes #115

# Type of Change:
- Code

# Checklist:
- [x] My PR follows the style guidelines of this project
- [x] I have performed a self-review of my own code or materials
- [x] My changes generate no new warnings

ramitsawhney27 commented:
@sammy1997 why is the label of not reviewed still on this even though I’ve reviewed it?

sara-02 commented:
@ramitsawhney27 We still have this as an enhancement to auto update the non-approved to approved label. https://github.com/systers/sysbot/issues/127
Until then we have to do it manually :)

sara-02 commented:
```bash
code/main_server.py
code/main_server.py:1 at module level:
        D100: Missing docstring in public module
code/main_server.py:254 in public function `get_stems`:
        D401: First line should be in imperative mood ('Perform', not 'Performs')
code/main_server.py:262 in public function `lemmatize_sent`:
        D401: First line should be in imperative mood ('Perform', not 'Performs')
3
    Fail
code/slack_functions.py
code/slack_functions.py:1 at module level:
        D100: Missing docstring in public module
code/slack_functions.py:139 in public function `assign_issue_slack`:
        D401: First line should be in imperative mood ('Assign', not 'Assigns')
code/slack_functions.py:258 in public function `send_message_thread`:
        D401: First line should be in imperative mood ('Send', not 'Sends')
code/slack_functions.py:265 in public function `send_message_ephemeral`:
        D401: First line should be in imperative mood ('Send', not 'Sends')
code/slack_functions.py:272 in public function `open_issue_slack`:
        D401: First line should be in imperative mood ('Open', not 'Opens')
code/slack_functions.py:303 in public function `get_detailed_profile`:
        D401: First line should be in imperative mood ('Fetch', not 'Fetches')
code/slack_functions.py:313 in public function `get_github_username_profile`:
        D401: First line should be in imperative mood ('Fetch', not 'Fetches')
code/slack_functions.py:325 in public function `slack_team_name_reply`:
        D401: First line should be in imperative mood ('Reply', not 'Replies')
code/slack_functions.py:364 in public function `handle_message_answering`:
        D401: First line should be in imperative mood ('Handle', not 'Handles')
code/slack_functions.py:430 in public function `check_is_question`:
        D401: First line should be in imperative mood ('Check', not 'Checks')
10
    Fail
code/github_functions.py
code/github_functions.py:1 at module level:
        D100: Missing docstring in public module
code/github_functions.py:42 in public function `send_github_invite`:
        D401: First line should be in imperative mood ('Send', not 'Sends')
code/github_functions.py:94 in public function `issue_assign`:
        D401: First line should be in imperative mood ('Assign', not 'Assigns')
code/github_functions.py:143 in public function `check_multiple_issue_claim`:
        D401: First line should be in imperative mood ('Check', not 'Checks')
code/github_functions.py:173 in public function `get_issue_author`:
        D401: First line should be in imperative mood ('Fetch', not 'Fetches')
code/github_functions.py:182 in public function `check_approved_tag`:
        D401: First line should be in imperative mood ('Check', not 'Checks')
code/github_functions.py:197 in public function `unassign_issue`:
        D401: First line should be in imperative mood ('Remove', not 'Removes')
code/github_functions.py:208 in public function `close_pr`:
        D401: First line should be in imperative mood ('Close', not 'Closes')
code/github_functions.py:220 in public function `check_issue_template`:
        D401: First line should be in imperative mood ('Validate', not 'Validates')
code/github_functions.py:233 in public function `are_issue_essential_components_present`:
        D401: First line should be in imperative mood ('Check', not 'Checks')
code/github_functions.py:277 in public function `list_open_prs_from_repo`:
        D401: First line should be in imperative mood ('List', not 'Lists')
code/github_functions.py:299 in public function `check_pr_template`:
        D401: First line should be in imperative mood ('Check', not 'Checks')
12
    Fail
code/request_urls.py
code/request_urls.py:1 at module level:
        D100: Missing docstring in public module
1
    Fail
code/topic_extractor.py
code/topic_extractor.py:1 at module level:
        D100: Missing docstring in public module
code/topic_extractor.py:31 in public class `NPExtractor`:
        D101: Missing docstring in public class
code/topic_extractor.py:33 in public method `__init__`:
        D107: Missing docstring in __init__
code/topic_extractor.py:38 in public method `tokenize_sentence`:
        D102: Missing docstring in public method
code/topic_extractor.py:44 in public method `normalize_tags`:
        D102: Missing docstring in public method
code/topic_extractor.py:60 in public method `extract`:
        D102: Missing docstring in public method
6
    Fail
code/__init__.py
code/__init__.py:1 at module level:
        D104: Missing docstring in public package
1
    Fail
code/auth_credentials.py
code/auth_credentials.py:1 at module level:
        D100: Missing docstring in public module
1
    Fail
code/messages.py
code/messages.py:1 at module level:
        D100: Missing docstring in public module
1
    Fail
code/dictionaries.py
code/dictionaries.py:1 at module level:
        D100: Missing docstring in public module
1
    Fail
tests/test_slack_functions.py
tests/test_slack_functions.py:1 at module level:
        D100: Missing docstring in public module
tests/test_slack_functions.py:13 in public class `TestSlackFunctions`:
        D204: 1 blank line required after class docstring (found 0)
tests/test_slack_functions.py:15 in public method `test_get_github_username_profile`:
        D210: No whitespaces allowed surrounding docstring text
tests/test_slack_functions.py:76 in public method `test_approve_issue_label_slack`:
        D400: First line should end with a period (not ')')
4
    Fail
tests/test_github_functions.py
tests/test_github_functions.py:1 at module level:
        D100: Missing docstring in public module
tests/test_github_functions.py:15 in public class `TestGithubFunctions`:
        D204: 1 blank line required after class docstring (found 0)
2
    Fail
tests/setup_data.py
tests/setup_data.py:1 at module level:
        D100: Missing docstring in public module
1
    Fail
tests/__init__.py
tests/__init__.py:1 at module level:
        D104: Missing docstring in public package
1
    Fail
tests/test_main_server.py
tests/test_main_server.py:1 at module level:
        D100: Missing docstring in public module
tests/test_main_server.py:11 in public class `TestMainServer`:
        D204: 1 blank line required after class docstring (found 0)
2
    Fail
Documentation strings should be added and/or fixed in 14 source files out of 14 files

```

GeVic commented:
@sara02 I'll sure add params and return values. @ramitsawhney27 What about the test functions? Is that also supposed to be changed?

sammy1997 commented:
Also @GeVic please edit the commit message as per the commit message guide https://github.com/systers/sysbot/wiki/Commit-Message-Style-Guide . That way we can maintain a standard for the messages :)

sara-02 commented:
@Gevic for test you can just say `Mock Test <function_name>() for github/slack`

sara-02 commented:
You need not elaborate test doc strings, the equivalent function doc string should be comprehensive.

GeVic commented:
@ramitsawhney27 @sara-02 Before proceeding further and just to be sure. Check the below screenshot and comment on it if it's ok to proceed.

![pic](https://user-images.githubusercontent.com/29516629/43679672-8675331c-9846-11e8-90b5-5faf407aae27.PNG)

GeVic commented:
@sara-02 I don't mind changing the format but this was the format mentioned and suggested elsewhere.

sara-02 commented:
@GeVic Can you point to that source? I have never come across this format :/, maybe I am missing something.

GeVic commented:
@sara-02  Although I changed the format as requested but still as you asked below are some of the links I came across which indicated the format I added doc-string in in the first place.
http://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html
https://www.geeksforgeeks.org/python-docstrings/

sammy1997 commented:
@GeVic There are 5 commits in the PR. Can you squash them into 1? It's not a good practice to have more than 1/2 commits in a PR. :)

GeVic commented:
@sammy1997  I already did.

sara-02 commented:
```
code/main_server.py
code/main_server.py:1 at module level:
        D100: Missing docstring in public module
code/main_server.py:25 in public function `collect_unreviewed_prs`:
        D200: One-line docstring should fit on one line with quotes (found 3)
code/main_server.py:45 in public function `home`:
        D205: 1 blank line required between summary line and description (found 0)
code/main_server.py:54 in public function `github_hook_receiver_function`:
        D205: 1 blank line required between summary line and description (found 0)
code/main_server.py:184 in public function `slack_hook_receiver_function`:
        D205: 1 blank line required between summary line and description (found 0)
code/main_server.py:225 in public function `invite`:
        D205: 1 blank line required between summary line and description (found 0)
code/main_server.py:240 in public function `slack_approval_receiver`:
        D205: 1 blank line required between summary line and description (found 0)
code/main_server.py:252 in public function `slack_assign_receiver`:
        D205: 1 blank line required between summary line and description (found 0)
code/main_server.py:264 in public function `slack_claim_receiver`:
        D205: 1 blank line required between summary line and description (found 0)
code/main_server.py:276 in public function `open_issue_receiver`:
        D205: 1 blank line required between summary line and description (found 0)
code/main_server.py:288 in public function `help_command`:
        D205: 1 blank line required between summary line and description (found 0)
code/main_server.py:300 in public function `get_stems`:
        D205: 1 blank line required between summary line and description (found 0)
code/main_server.py:311 in public function `lemmatize_sent`:
        D205: 1 blank line required between summary line and description (found 0)
code/main_server.py:311 in public function `lemmatize_sent`:
        D401: First line should be in imperative mood ('Perform', not 'Performs')
14
    Fail
code/slack_functions.py
code/slack_functions.py:1 at module level:
        D100: Missing docstring in public module
code/slack_functions.py:25 in public function `dm_new_users`:
        D205: 1 blank line required between summary line and description (found 0)
code/slack_functions.py:25 in public function `dm_new_users`:
        D400: First line should end with a period (not 'M')
code/slack_functions.py:55 in public function `is_maintainer_comment`:
        D205: 1 blank line required between summary line and description (found 0)
code/slack_functions.py:77 in public function `approve_issue_label_slack`:
        D205: 1 blank line required between summary line and description (found 0)
code/slack_functions.py:77 in public function `approve_issue_label_slack`:
        D400: First line should end with a period (not 'm')
code/slack_functions.py:134 in public function `check_newcomer_requirements`:
        D205: 1 blank line required between summary line and description (found 0)
code/slack_functions.py:134 in public function `check_newcomer_requirements`:
        D401: First line should be in imperative mood ('Check', not 'Checks')
code/slack_functions.py:161 in public function `assign_issue_slack`:
        D205: 1 blank line required between summary line and description (found 0)
code/slack_functions.py:161 in public function `assign_issue_slack`:
        D400: First line should end with a period (not 'm')
code/slack_functions.py:212 in public function `claim_issue_slack`:
        D205: 1 blank line required between summary line and description (found 0)
code/slack_functions.py:212 in public function `claim_issue_slack`:
        D400: First line should end with a period (not 'e')
code/slack_functions.py:282 in public function `send_message_to_channels`:
        D205: 1 blank line required between summary line and description (found 0)
code/slack_functions.py:282 in public function `send_message_to_channels`:
        D401: First line should be in imperative mood ('Send', not 'Sends')
code/slack_functions.py:297 in public function `send_message_thread`:
        D205: 1 blank line required between summary line and description (found 0)
code/slack_functions.py:297 in public function `send_message_thread`:
        D401: First line should be in imperative mood ('Send', not 'Sends')
code/slack_functions.py:310 in public function `send_message_ephemeral`:
        D205: 1 blank line required between summary line and description (found 0)
code/slack_functions.py:310 in public function `send_message_ephemeral`:
        D400: First line should end with a period (not 'e')
code/slack_functions.py:324 in public function `open_issue_slack`:
        D205: 1 blank line required between summary line and description (found 0)
code/slack_functions.py:324 in public function `open_issue_slack`:
        D400: First line should end with a period (not 'k')
code/slack_functions.py:359 in public function `get_detailed_profile`:
        D205: 1 blank line required between summary line and description (found 0)
code/slack_functions.py:359 in public function `get_detailed_profile`:
        D400: First line should end with a period (not 's')
code/slack_functions.py:374 in public function `get_github_username_profile`:
        D205: 1 blank line required between summary line and description (found 0)
code/slack_functions.py:374 in public function `get_github_username_profile`:
        D401: First line should be in imperative mood ('Extract', not 'Extracts')
code/slack_functions.py:390 in public function `slack_team_name_reply`:
        D205: 1 blank line required between summary line and description (found 0)
code/slack_functions.py:390 in public function `slack_team_name_reply`:
        D401: First line should be in imperative mood ('Reply', not 'Replies')
code/slack_functions.py:436 in public function `handle_message_answering`:
        D205: 1 blank line required between summary line and description (found 0)
code/slack_functions.py:496 in public function `answer_keyword_faqs`:
        D205: 1 blank line required between summary line and description (found 0)
code/slack_functions.py:518 in public function `check_is_question`:
        D205: 1 blank line required between summary line and description (found 0)
code/slack_functions.py:518 in public function `check_is_question`:
        D401: First line should be in imperative mood ('Check', not 'Checks')
code/slack_functions.py:533 in public function `luis_classifier`:
        D205: 1 blank line required between summary line and description (found 0)
code/slack_functions.py:533 in public function `luis_classifier`:
        D400: First line should end with a period (not 'r')
32
    Fail
code/github_functions.py
code/github_functions.py:1 at module level:
        D100: Missing docstring in public module
code/github_functions.py:18 in public function `label_opened_issue`:
        D205: 1 blank line required between summary line and description (found 0)
code/github_functions.py:46 in public function `send_github_invite`:
        D205: 1 blank line required between summary line and description (found 0)
code/github_functions.py:46 in public function `send_github_invite`:
        D400: First line should end with a period (not 'e')
code/github_functions.py:66 in public function `issue_comment_approve_github`:
        D205: 1 blank line required between summary line and description (found 0)
code/github_functions.py:100 in public function `github_pull_request_label`:
        D205: 1 blank line required between summary line and description (found 0)
code/github_functions.py:117 in public function `issue_assign`:
        D205: 1 blank line required between summary line and description (found 0)
code/github_functions.py:136 in public function `check_assignee_validity`:
        D205: 1 blank line required between summary line and description (found 0)
code/github_functions.py:151 in public function `github_comment`:
        D205: 1 blank line required between summary line and description (found 0)
code/github_functions.py:170 in public function `issue_claim_github`:
        D205: 1 blank line required between summary line and description (found 0)
code/github_functions.py:193 in public function `check_multiple_issue_claim`:
        D205: 1 blank line required between summary line and description (found 0)
code/github_functions.py:215 in public function `open_issue_github`:
        D205: 1 blank line required between summary line and description (found 0)
code/github_functions.py:215 in public function `open_issue_github`:
        D400: First line should end with a period (not 'd')
code/github_functions.py:240 in public function `get_issue_author`:
        D205: 1 blank line required between summary line and description (found 0)
code/github_functions.py:255 in public function `check_approved_tag`:
        D205: 1 blank line required between summary line and description (found 0)
code/github_functions.py:276 in public function `unassign_issue`:
        D205: 1 blank line required between summary line and description (found 0)
code/github_functions.py:294 in public function `close_pr`:
        D205: 1 blank line required between summary line and description (found 0)
code/github_functions.py:312 in public function `check_issue_template`:
        D205: 1 blank line required between summary line and description (found 0)
code/github_functions.py:332 in public function `are_issue_essential_components_present`:
        D205: 1 blank line required between summary line and description (found 0)
code/github_functions.py:380 in public function `list_open_prs_from_repo`:
        D205: 1 blank line required between summary line and description (found 0)
code/github_functions.py:407 in public function `check_pr_template`:
        D205: 1 blank line required between summary line and description (found 0)
code/github_functions.py:449 in public function `label_list_issue`:
        D103: Missing docstring in public function
code/github_functions.py:471 in public function `pr_reviewed_label`:
        D103: Missing docstring in public function
23
    Fail
code/request_urls.py
code/request_urls.py:1 at module level:
        D100: Missing docstring in public module
1
    Fail
code/topic_extractor.py
code/topic_extractor.py:1 at module level:
        D100: Missing docstring in public module
code/topic_extractor.py:31 in public class `NPExtractor`:
        D101: Missing docstring in public class
code/topic_extractor.py:33 in public method `__init__`:
        D107: Missing docstring in __init__
code/topic_extractor.py:38 in public method `tokenize_sentence`:
        D102: Missing docstring in public method
code/topic_extractor.py:44 in public method `normalize_tags`:
        D102: Missing docstring in public method
code/topic_extractor.py:60 in public method `extract`:
        D102: Missing docstring in public method
6
    Fail
code/__init__.py
code/__init__.py:1 at module level:
        D104: Missing docstring in public package
1
    Fail
code/auth_credentials.py
code/auth_credentials.py:1 at module level:
        D100: Missing docstring in public module
1
    Fail
code/messages.py
code/messages.py:1 at module level:
        D100: Missing docstring in public module
1
    Fail
code/dictionaries.py
code/dictionaries.py:1 at module level:
        D100: Missing docstring in public module
1
    Fail
tests/test_slack_functions.py
tests/test_slack_functions.py:1 at module level:
        D100: Missing docstring in public module
tests/test_slack_functions.py:13 in public class `TestSlackFunctions`:
        D204: 1 blank line required after class docstring (found 0)
tests/test_slack_functions.py:15 in public method `test_get_github_username_profile`:
        D210: No whitespaces allowed surrounding docstring text
tests/test_slack_functions.py:76 in public method `test_approve_issue_label_slack`:
        D400: First line should end with a period (not ')')
4
    Fail
tests/test_github_functions.py
tests/test_github_functions.py:1 at module level:
        D100: Missing docstring in public module
tests/test_github_functions.py:17 in public class `TestGithubFunctions`:
        D204: 1 blank line required after class docstring (found 0)
tests/test_github_functions.py:166 in public method `test_label_list_issue`:
        D210: No whitespaces allowed surrounding docstring text
tests/test_github_functions.py:173 in public method `test_pr_reviewed_label`:
        D210: No whitespaces allowed surrounding docstring text
4
    Fail
tests/setup_data.py
tests/setup_data.py:1 at module level:
        D100: Missing docstring in public module
1
    Fail
tests/__init__.py
tests/__init__.py:1 at module level:
        D104: Missing docstring in public package
1
    Fail
tests/test_main_server.py
tests/test_main_server.py:1 at module level:
        D100: Missing docstring in public module
tests/test_main_server.py:11 in public class `TestMainServer`:
        D204: 1 blank line required after class docstring (found 0)
2
    Fail
Documentation strings should be added and/or fixed in 14 source files out of 14 files
```

sara-02 commented:
Also, you need a one line doc string at the start of every `.py` file

sara-02 commented:
@GeVic Great work, almost there :)

sara-02 commented:
@GeVic Still some issues got missed :see_no_evil: 
```
code/slack_functions.py
code/slack_functions.py:304 in public function `send_message_thread`:
        D401: First line should be in imperative mood ('Send', not 'Sends')
code/slack_functions.py:539 in public function `luis_classifier`:
        D205: 1 blank line required between summary line and description (found 0)
code/slack_functions.py:539 in public function `luis_classifier`:
        D400: First line should end with a period (not '-')
3
    Fail
code/github_functions.py
code/github_functions.py:339 in public function `are_issue_essential_components_present`:
        D205: 1 blank line required between summary line and description (found 0)
code/github_functions.py:339 in public function `are_issue_essential_components_present`:
        D400: First line should end with a period (not 'y')
code/github_functions.py:388 in public function `list_open_prs_from_repo`:
        D205: 1 blank line required between summary line and description (found 0)
code/github_functions.py:388 in public function `list_open_prs_from_repo`:
        D400: First line should end with a period (not 'e')
4
    Fail
code/request_urls.py
0
    Pass
code/topic_extractor.py
code/topic_extractor.py:47 in public method `normalize_tags`:
        D400: First line should end with a period (not ')')
code/topic_extractor.py:63 in public method `extract`:
        D202: No blank lines allowed after function docstring (found 1)
2
    Fail
code/__init__.py
code/__init__.py:1 at module level:
        D400: First line should end with a period (not 'y')
1
    Fail
code/auth_credentials.py
code/auth_credentials.py:1 at module level:
        D400: First line should end with a period (not ')')
1
    Fail
code/messages.py
0
    Pass
code/dictionaries.py
WARNING: Error in file code/dictionaries.py: Cannot parse file.
1
    Fail
tests/test_slack_functions.py
tests/test_slack_functions.py:1 at module level:
        D100: Missing docstring in public module
tests/test_slack_functions.py:13 in public class `TestSlackFunctions`:
        D204: 1 blank line required after class docstring (found 0)
tests/test_slack_functions.py:76 in public method `test_approve_issue_label_slack`:
        D400: First line should end with a period (not ')')
3
    Fail
tests/test_github_functions.py
0
    Pass
tests/setup_data.py
0
    Pass
tests/__init__.py
tests/__init__.py:1 at module level:
        D400: First line should end with a period (not 'y')
1
    Fail
tests/test_main_server.py
0
    Pass
Documentation strings should be added and/or fixed in 8 source files out of 14 files

```

sara-02 commented:
Also, please resolve the Merge conflicts by rebasing your branch.

sammy1997 commented:
@GeVic please pull the new changes into your branch and solve the merge conflicts

GeVic commented:
@sammy1997 @sara-02 Okay, for this sole reason I used doc-formater. Because this PR becomes more of labour works which is supposed to be intresting instead. 
Also, the current PR might take some time because of the nature of it's work, so, please consider the delay.

sara-02 commented:
@GeVic kudos to your effort. It is not easy to provide a good fix for Python linting as there are so many formats. We are justing to make sure we cover as much ground as possible. Your work will make the life of future contributors easy, trust me on that. 
You can take your time to get this covered. :)

GeVic commented:
@sara-02 Don't worry, I'll resolve the conflicts asap.

sammy1997 commented:
@GeVic whats the status on this PR?

GeVic commented:
@sammy1997 @sara-02 I wasn't able to work on the same for quite a some time now and so instead of giving reasons I would try and finish it up today itself.

GeVic commented:
@sara-02 Once you review the PR, I'll squash up the commits.

sara-02 commented on .coveragerc:
```diff
@@ -2,5 +2,8 @@
 omit =
     */tests/*
     code/messages.py
+<<<<<<< HEAD
```
This needs to be removed.

sara-02 commented on .coveragerc:
```diff
@@ -2,5 +2,8 @@
     */tests/*
     code/messages.py
+<<<<<<< HEAD
+=======
```
This as well.

sara-02 commented on .coveragerc:
```diff
@@ -2,5 +2,8 @@
+<<<<<<< HEAD
+=======
     code/dictionaries_test.py
+>>>>>>> 02fbaf1413cdb49c4457bf8bc144a4ae5ce62397
```
This too.

GeVic commented:
@sara-02 conflicting lines removed. Do have a look at the PR again please.

sara-02 commented:
@GeVic just squash the commits. We are good to merge this :)
